src/main/java/io/appium/java_client/AppiumFluentWait.java

```diff
@@ -201,9 +216,17 @@ public AppiumFluentWait<T> withPollingStrategy(Function<IterationInfo, Duration>
     @Override
     public <V> V until(Function<? super T, V> isTrue) {
         final Instant start = getClock().instant();
-        final Instant end = getClock().instant().plus(getTimeout());
+        final Instant end = start.plus(getTimeout()).plus(pollDelay);
```

Comment:
we are already sleeping in performIteration. Why do we need to make another delay here?

Reply:
In the way I see the `pollDelay`, it should delay the moment when we start to verify the condition. 

If we don't add this time to the `end` instant, it will steal time used for verifying the condition. Paradoxically, if the `pollDelay >= timeout`, the verification will never start, and the process will throw `TimeoutException`.

This is my interpretation and the way I wanted to implement it. But I am open to suggestions.

Reply:
makes sense. Then probably mention the above in a comment to improve the readability

Reply:
Done